src/ui/templates/components/single-radio.njk

```diff
@@ -0,0 +1,90 @@
+{% from 'govuk/components/button/macro.njk' import govukButton %}
+{% import './yes-no-radio-buttons.njk' as yesNoRadioButtons %}
+{% import '../components/submit-button.njk' as submitButton %}
+{% from "govuk/components/details/macro.njk" import govukDetails %}
+
+{% macro render(params) %}
+
+  {% set fieldId = params.fieldId %}
+  {% set legendText = params.legendText %}
+  {% set hintText = params.hintText %}
+  {% set hasHintHtml = params.hasHintHtml %}
+  {% set submittedAnswer = params.submittedAnswer %}
+  {% set horizontalRadios = params.horizontalRadios %}
+  {% set errorMessage = params.errorMessage %}
+  {% set conditionalYes = params.conditionalYes %}
+  {% set conditionalNo = params.conditionalNo %}
+  {% set continueButtonText = params.continueButtonText %}
+  {% set saveAndBackButtonText = params.saveAndBackButtonText %}
+  {% set saveAndBackUrl = params.saveAndBackUrl %}
+  {% set hasSaveAndBack = params.hasSaveAndBack %}
+  {% set hasCustomContent = params.hasCustomContent %}
+  {% set customContentHTML = params.customContentHTML %}
+  {% set CONTENT_STRINGS = params.CONTENT_STRINGS %}
+  {% set validationErrors = params.validationErrors %}
+  {% set submittedValues = params.submittedValues %}
+  {% set FIELDS = params.FIELDS %}
+  {% set application = params.application %}
+  {% set legendClass = params.legendClass %}
+
+  {% if conditionalYes %}
+    {% set conditionalYesHtml %}
+      {% include conditionalYes %}
+    {% endset %}
+  {% endif %}
+
+  {% if conditionalNo %}
+    {% set conditionalNoHtml %}
+      {% include conditionalNo %}
+    {% endset %}
+  {% endif %}
+
+  {% if hasHintHtml %}
+    {% set hintHtml %}
+      {% include hasHintHtml %}
+    {% endset %}
+  {% endif %}
```

Comment:
include `hasHintHtml`?

What's the difference between this and` the `hintHtml` param passed to `yesNoRadioButtons`?

Seems we have 2x things that render custom hint HTML, `hasHintHtml` and `hintHtml`

Reply:
So I need something set as I have for hintHtml or else it just prints a string.  hasHintHtml has the path for the includes

Reply:
Could we rename `hasHintHtml` to `hintHtmlPartial`? This reads a bit strange:

```js
{% include hasHintHtml %}
```

Reply:
These variables aren't used, can be removed
- `CONTENT_STRINGS`
- `submittedValues` 
- `validationErrors`

Also `legendText` is set twice

Reply:
still need as in partial